zebra/zebra_vty.c

```diff
@@ -433,13 +436,27 @@ static void zebra_show_ip_route_opaque(struct vty *vty, struct route_entry *re,
 			vty_out(vty, "    Opaque Data: %s",
 				(char *)re->opaque->data);
 		break;
-	case ZEBRA_ROUTE_BGP:
-		if (json)
-			json_object_string_add(json, "asPath",
-					       (char *)re->opaque->data);
-		else
-			vty_out(vty, "    AS-Path: %s",
-				(char *)re->opaque->data);
+	case ZEBRA_ROUTE_BGP: {
+		memcpy(&bzo, re->opaque->data, sizeof(bzo));
```

Comment:
hmm - but we have `re->opaque->length` available? the assumption is that bgp will never send anything but this struct in its entirety, but it'd still look better to avoid looking beyond the bytes that we actually got via zapi.